src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1684,7 +1684,9 @@ static int client_receive_message_udp(
         assert(client);
 
         buflen = next_datagram_size_fd(fd);
-        if (buflen < 0)
+        if (buflen == -ENETDOWN)
+                return 0;
+        else if (buflen < 0)
```

Comment:
hmm, can you drop the `else` here? In most our other code we wouldn't put it, because we emphasize "exit early on failure" semantics, and given that the first branch returned already the `else` is redundant, and a simple `if` suffices.

(this doesn't matter at all though, just helps my ocd)